src/common/libsubprocess/subprocess.c

```diff
@@ -829,22 +848,22 @@ int flux_subprocess_read (flux_subprocess_t *p,
                           const char *stream,
                           const char **bufp)
 {
-    return subprocess_read (p, stream, bufp, false, false, false, NULL);
+    return subprocess_read (p, stream, bufp, false, false, false, true, NULL);
```

Comment:
This method of reducing duplicated code with a common function with booleans that alter its behavior is hard to follow.  Instead of adding one more boolean to this function, how about adding a check to `flux_subprocess_read_line()` et al like this:
```c
if (p && (p->flags & FLUX_SUBPROCESS_FLAGS_LOCAL_UNBUF)) {
    errno = EINVAL;
    return -1;
}
```